teachers.rb

```diff
@@ -2,8 +2,10 @@
 
 # subclass for the person
 class Teacher < Person
+  attr_reader :specialization
+
   def initialize(id, age, specialization, name: 'Unknown', parent_permission: true)
-    super(id, age, name: name, parent_permission: parent_permission)
+    super(id, age, name, parent_permission)
     @specialization = specialization
   end
```

Comment:
As requested by the previous reviewer, please structure this as well by removing the `id` as the example codes provided to you above 👍